### PR TITLE
.mvn/excludes.txt updates

### PR DESCRIPTION
**Pull request by ppalaga**
Replaces https://github.com/jboss-fuse/camel-quarkus/pull/48

* Backport of some important community commits
* Update .mvn/excludes.txt for 2.2.0 603a429
* Add extensions required by Camel K Runtime 727901a
* Check the consistency of camel-quarkus-product-source.json with the source tree fc1f987

Still todo: 
* Exclude the non-productized modules as a regular commit
* Split some tests so that they can be kept in the prod module tree without missing dependencies

### CODE REVIEW

**Comment by cunningt** [comment]
Do we need camel-quarkus-support-webhook?    camel-k-runtime seems to require it, I'm not sure if it is actually needed or we can remove something there.

I tried the excludes list here with camel-k-runtime, it was missing webhook modules, and the dsl modules but https://issues.redhat.com/browse/ENTESB-17277 seems to suggest that we can disable the use of the dsl modules possibly.

File: .mvn/excludes.txt [hunk 1]
```diff
@@ -1127,9 +1116,6 @@
 :camel-quarkus-xj-deployment
 :camel-quarkus-xj-integration-test
 :camel-quarkus-xj-parent
-:camel-quarkus-xml-io-dsl
```
Comment (jamesnetherton) [inline .mvn/excludes.txt]: Is the XML DSL definately required? Neither CQ or Camel-K product support it AFAIK.

**Comment by ppalaga** [comment]
> Do we need camel-quarkus-support-webhook?

I thought it will be included as a transitive dependency of some supported extension. Let me check why it flow away.

File: .mvn/excludes.txt (hunk 1 above)
Comment (ppalaga) [reply .mvn/excludes.txt]: I included it based on @lburgazzoli 's statement

> camel-quarkus-yaml-dsl, camel-quarkus-java-joor-dsl, camel-quarkus-xml-io-dsl  are the support extension for the languages we support in camel-k

Comment (ppalaga) [reply .mvn/excludes.txt]: I can remove it again if there is a strong enough evidence that XML is not supported in Camel K

Comment (jamesnetherton) [reply .mvn/excludes.txt]: Docs only mention Java & YAML:

https://access.redhat.com/documentation/en-us/red_hat_integration/2021.q3/html/getting_started_with_camel_k/developing-and-running-camel-k-integrations

Comment (lburgazzoli) [reply .mvn/excludes.txt]: Yes, I think it was my mistake because as a long time camel rider, I assumed XML and Java were supported by default.

**Comment by ppalaga** [comment]
31a0893 - `camel-quarkus-support-webhook` added to `camel-quarkus-product-source.json` explicitly. Only telegram depends on it so it cannot be included transitively.

**Comment by cunningt** [comment]
Is there a JIRA or github issue regarding modularizing/disabling/removing the use of the camel-quarkus dsl from camel-k-runtime?    We have a branch (https://github.com/jboss-fuse/camel-k-runtime/tree/camel-k-runtime-1.9.0-branch).

File: .mvn/excludes.txt (hunk 1 above)
Comment (ppalaga) [reply .mvn/excludes.txt]: OK, let's exclude XML

**Comment by ppalaga** [comment]
0f86855: removed XML DSL

**Review by cunningt — APPROVED** [review]

**Review by jamesnetherton — APPROVED** [review]

**Comment by jamesnetherton** [comment]
@ppalaga Is this good to merge?

**Pull request merged**